app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt

```diff
@@ -482,28 +483,43 @@ public class MainActivity : AppCompatActivity(), ViewController, Router.Callback
         startActivityForResult(intent, requestCodeSearchResults)
     }
 
-    override fun showRoutingMode() {
+    override fun showRoutingMode(feature: Feature) {
```

Comment:
I like having the presenter pass back the destination feature here each time. As a design decision, what do you think about eliminating the properties `route